lib/promscrape/scrapework.go

```diff
@@ -463,13 +463,16 @@ func (sw *scrapeWork) processDataOneShot(scrapeTimestamp, realTimestamp int64, b
 	wc := writeRequestCtxPool.Get(sw.prevLabelsLen)
 	lastScrape := sw.loadLastScrape()
 	bodyString := bytesutil.ToUnsafeString(body)
-	areIdenticalSeries := sw.areIdenticalSeries(lastScrape, bodyString)
 	if err != nil {
 		up = 0
 		scrapesFailed.Inc()
+		// body may have non-empty value in case of error
+		// explicitly set it to empty to properly compute areIdenticalSeries later
+		bodyString = ""
 	} else {
 		wc.rows.UnmarshalWithErrLogger(bodyString, sw.logError)
```

Comment:
shall we move `bodyString := bytesutil.ToUnsafeString(body)` here? and remove 471 line?

Reply:
and we can remove L:490-492 as well

Reply:
at L:490-492 we cannot remove check, since actual request parsing may change `up` state:
```
wc.resetNoRows()
up = 0
```

Reply:
I think, it's better to follow the same logic, if there is some error and `up == 0`, nullify body.

 Otherwise, this code may look like a typo:

```
var bodyString string
if err != nil {
	up = 0
}else{
   bodyString = bytesutil.ToUnsafeString(body)
}
```

 If I'm not aware of this bug, I'd move `bodyString := bytesutil.ToUnsafeString(body)` and remove `var bodyString` declaration.